libraries/lib-uuid/Uuid.h

```diff
@@ -0,0 +1,51 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file Uuid.h
+ @brief Declare a class to generate and parse UUIDs
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <string>
+
+namespace audacity
+{
+
+class UUID_API Uuid final
+{
+public:
+   using Bytes = std::array<uint8_t, 16>;
+
+   Uuid();
+   explicit Uuid(const Bytes& data) noexcept;
+
+   static Uuid Generate();
+   static Uuid FromString(const std::string& str);
+
+   bool isNil() const noexcept;
```

Comment:
Ah, one more thing.

Why not negate the return value, and call this explicit operator bool ?

Reply:
That could make it more concise to declare a Uuid variable within `if(...)` and narrow its scope.